### DISCUSSION TITLE
how to fix this error

### DISCUSSION THREAD

**Discussion by explainmetech** [mesutpiskin/keycloak-2fa-email-authenticator#41] [General]
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.13.0:compile (default-compile) on project keycloak-2fa-email-authenticator: Fatal error compiling: java.lang.NoSuchFieldError: Class com.sun.tools.javac.tree.JCTree$JCImport does not have member field 'com.sun.tools.javac.tree.JCTree qualid' -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

**Comment by bobbabuilder** [comment]
https://stackoverflow.com/questions/77171270/compilation-error-after-upgrading-to-jdk-21-nosuchfielderror-jcimport-does-n

**Comment by mesutpiskin** [comment]
This error is a **JDK version incompatibility** — it is not specific to this plugin. It occurs when you try to build with **JDK 23 or later** against an older version of the Maven compiler plugin or Lombok.

The project requires **JDK 21**. Please make sure you are building with JDK 21:
```bash
java -version   # should print openjdk 21.x.x
```

If you are just looking to use the plugin (not build from source), download the pre-built JAR directly from the [latest release](https://github.com/mesutpiskin/keycloak-2fa-email-authenticator/releases/latest) or from Maven Central:
```xml
<dependency>
  <groupId>io.github.mesutpiskin</groupId>
  <artifactId>keycloak-2fa-email-authenticator</artifactId>
  <version>26.3.1-KC26.6.1</version>
</dependency>
```
